experimental/dispatch_profiler/matmul.py

```diff
@@ -0,0 +1,590 @@
+
+    # Save the expected result as an np.array.
+    np.save(os.path.join(output_dir, self.reference_filename_result),\
+             np.array(result, dtype = self.dtype_result))
+
+
+###############################################################################
+class MatmulOperationLauncher:
+  """Launches the compilation and execution of the matmul operation.
+  MatmulOperationLauncher class has the following responsibilities:
+    1) Launches compilation of the matmul for a verification or profiling runs.
+    2) Launches the verification by running the IREE compiled matmul operation
+       and the python nmpy.matmul.
+    3) Launches the profiling by running the IREE compiled matmul operation.
+  """
+
+  def __init__(self, args, operation):
+    self.operation = operation
+
+    # Variables from top-level argparse.
+    self.generated_path = os.path.join(args.build_dir, 'generated',
+                                       args.mlir_dialect)
+    self.device = args.device
+    self.benchmark_dispatch_repeat_count = args.batch_size
+    self.batch_size = args.batch_size
+    self.benchmark_repetitions = args.benchmark_repetitions
+    self.verbose = False if args.verbose in ['False', 'false', '0'] else True
+
+    # Additional paths.
+    self.matmul_path = os.path.join(self.generated_path, 'matmul')
+    self.operation_path = os.path.join(self.matmul_path, operation.name())
+    self.source_mlir_file = os.path.join(self.operation_path,
+                                         operation.name() + '.mlir')
+
+    # path to iree-compile tool. (for compiling the input mlir file to vmfb)
+    self.iree_compile_path = os.path.join(args.build_dir, 'tools/iree-compile')
+    self.force_compile = False if args.force_compile in ['False', 'false', '0'
+                                                        ] else True
+
+    # path to iree-benchmark-module tool. (for performance benchmarking and profiling)
+    self.iree_benchmark_module_path = os.path.join(
+        args.build_dir, 'tools/iree-benchmark-module')
+
+    # path to iree-run-module tool. (for verification)
+    self.iree_run_module_path = os.path.join(args.build_dir,
+                                             'tools/iree-run-module')
+
+    # output vmfb files for the operation.
+    self.vmfb_verify_file = os.path.join(self.operation_path,
+                                         self.operation.name() + '_verify.vmfb')
+    self.vmfb_benchmark_file = os.path.join(
+        self.operation_path,
+        self.operation.name() + '_benchmark.vmfb')
+
+  def compile(self, compilation_mode):
+    """Compiles the matmul operation to a vmfb file for profiling."""
+    cmd_template = \
+    "${iree_compile_path} ${source_mlir_file} --iree-hal-target-backends=${device} "\
+    "--iree-hal-cuda-llvm-target-arch=sm_80 --iree-hal-benchmark-dispatch-repeat-count=${benchmark-dispatch-repeat-count} "\
+    "--iree-codegen-llvmgpu-use-mma-sync=true -o ${vmfb_file}"
```

Comment:
Maybe we can just use the built-in format string?

```suggestion
    "{iree_compile_path} {source_mlir_file} --iree-hal-target-backends={device} "\
    "--iree-hal-cuda-llvm-target-arch=sm_80 --iree-hal-benchmark-dispatch-repeat-count={benchmark-dispatch-repeat-count} "\
    "--iree-codegen-llvmgpu-use-mma-sync=true -o {vmfb_file}".format(
        iree_compile_path=self.iree_compile_path,
        source_mlir_file=self.source_mlir_file,
        device=self.device,
        benchmark-dispatch-repeat-count=benchmark_dispatch_repeat_count,
        vmfb_file=vmfb_file
    )
```